### PR TITLE
Add Storage management library

### PR DESCRIPTION
**Pull request by pakrym**
This adds a prototype of generated storage management library along with the test.

### CODE REVIEW

File: sdk/storage/Azure.Management.Storage/src/FileServicesClient.cs [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// Licensed under the MIT License.
+
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure.Management.Storage
+{
+    public partial class FileServicesClient
+    {
+        public FileServicesClient(string subscriptionId, TokenCredential tokenCredential): this(subscriptionId, tokenCredential, StorageManagementClientOptions.Default)
```
Comment (pakrym) [inline sdk/storage/Azure.Management.Storage/src/FileServicesClient.cs]: Client constructors have to be manually defined at this time.

**Comment by pakrym** [comment]
@erich-wang @allenjzhang do you want me to merge this PR or close it?

File: eng/CodeGeneration.targets [hunk 2]
```diff
@@ -5,18 +5,20 @@
     <_AutoRestCoreVersion>3.0.6257</_AutoRestCoreVersion>
     <_AutoRestCSharpVersion>https://github.com/Azure/autorest.csharp/releases/download/3.0.0-dev.20200323.2/autorest-csharp-v3-3.0.0-dev.20200323.2.tgz</_AutoRestCSharpVersion>
     <_SupportsCodeGeneration Condition="'$(IsClientLibrary)' == 'true'">true</_SupportsCodeGeneration>
-    <AutoRestConfiguration Condition="'$(AutoRestConfiguration)' == ''">autorest.md</AutoRestConfiguration>
-
+    <_DefaultInputName>$(MSBuildProjectDirectory)/autorest.md</_DefaultInputName>
```
Comment (weshaggard) [inline eng/CodeGeneration.targets]: Would it better to just have the local autorest.md link to the remote config? Instead of passing it explicitly here?

Comment (pakrym) [reply eng/CodeGeneration.targets]: Autorest doesn't allow it, it has to have a single main input, this why I need to do this little dance.

File: eng/CodeGeneration.targets [hunk 3]
```diff
@@ -5,18 +5,20 @@
...
 
-    <_GenerateCode Condition="'$(_SupportsCodeGeneration)' == 'true' AND Exists('$(AutoRestConfiguration)')">true</_GenerateCode>
+    <_GenerateCode Condition="'$(_SupportsCodeGeneration)' == 'true' AND '$(AutorestInput)' != ''">true</_GenerateCode>
     <UsesJsonSerialization Condition="'$(UsesJsonSerialization)' == ''">true</UsesJsonSerialization>
   </PropertyGroup>
 
   <Target Name="GenerateCode" Condition="'$(_GenerateCode)' == 'true'" >
     <RemoveDir Directories="$(MSBuildProjectDirectory)/Generated"/>
-    <Exec Command="npx autorest@$(_AutoRestVersion) --version=$(_AutoRestCoreVersion) --input=$(AutoRestConfiguration) --use=$(_AutoRestCSharpVersion) --output-folder=$(MSBuildProjectDirectory) --title=$(RootNamespace) --namespace=$(RootNamespace) --shared-source-folder=$(_SharedCodeDirectory) --verbose" />
+    <Exec Command="npx autorest@$(_AutoRestVersion) --version=$(_AutoRestCoreVersion) $(AutorestInput) --require=$(AutorestConfiguration) $(AutorestAdditionalParameters) --use=$(_AutoRestCSharpVersion) --output-folder=$(MSBuildProjectDirectory) --title=$(RootNamespace) --namespace=$(RootNamespace) --shared-source-folder=$(_SharedCodeDirectory) --verbose" />
```
Comment (erich-wang) [inline eng/CodeGeneration.targets]: normally the namespace is specified in central configuration as well, e.g.
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/readme.csharp.md

Comment (pakrym) [reply eng/CodeGeneration.targets]: And it's wrong there for track 2 libraries.

Comment (pakrym) [reply eng/CodeGeneration.targets]: We are working on figuring out configuration for track 2 libraries.

Comment (erich-wang) [reply eng/CodeGeneration.targets:21]: Added one issue https://github.com/Azure/autorest.csharp/issues/604 for track purpose

**Pull request closed without merging**